Support out of tree dt bindings (#14521)
zephyrproject-rtos/zephyr: doc/application/index.rst

You can also define the ``SOC_ROOT`` variable in the application
:file:`CMakeLists.txt` file.

DeviceTree Definitions
======================

> Additional DeviceTree directory tree's, or DTS_ROOT's, can be added by

Reviewer: When the word is plural you don't use an apostrophe:
```suggestion
Additional DeviceTree directory trees, or DTS_ROOTs, can be added by
```

Author: Yeah, this I should have known ...

creating this directory tree:

    dts/bindings/
    dts/common/

> 	dts/arm/

Reviewer 2: spaces or tab? cf above lines

    include/

Where 'arm' is changed to the appropriate architecture. Each directory
is optional. The binding directory contains bindings and the other
directories contain files that can be included from DT sources.

Once the directory structure is in place, you can use it by specifying

> it's location through the ``DTS_ROOT`` CMake Cache variable::

Reviewer: it's is a contraction for "it is" (yes, English is inconsistent where possessive is normally indicated by adding "'s" (as in "David's book":
```suggestion
its location through the ``DTS_ROOT`` CMake Cache variable::
```

Author: Huh, okay, so it can be possessive for names, but not for 'it'.

Will try to remember, thank you.

   cmake -DDTS_ROOT=<path to dts root>

> You can also define the the variable in the application

Reviewer 2: ```suggestion
You can also define the variable in the application
```
